modules/llrt_crypto/Cargo.toml

```diff
@@ -30,6 +30,18 @@ uuid = { version = "1", default-features = false, features = [
 ] }
 uuid-simd = "0.8"
 
+hmac = "0.12"
+sha1 = "0.10"
+sha2 = "0.10"
+aes = "0.8"
+aes-gcm = "0.10"
+cbc = { version = "0.1", features = ["std"] }
+num-traits = "0.2"
+rsa = { version = "0.9", features = ["std", "sha2"], default-features = false }
+p256 = { version = "0.13", features = ["ecdh"] }
+p384 = "0.13"
+ctr = "0.9"
+
```

Comment:
The only thing is that ring is C code (requires a C compiler) where the others are rust code.

Reply:
Hmm... I'm thinking of replacing some of the ring parts, but even though the names are the same, the methods are different, so it's not going to work.

It looks like I'll need to spend some time on this...

Reply:
Take RSA for example.

```
// Cargo.toml
...
#rsa = { version = "0.9", features = ["std", "sha2"], default-features = false }
```

Then, if you replace `use rsa::...` with `use ring::rsa::...`, you will get an error saying that the import cannot be resolved.

<img width="589" alt="image" src="https://github.com/user-attachments/assets/ce9c6267-2754-4d07-a395-9fc9c8f4f3be">

If the command doesn't exist, do we need to look for an alias? If it still doesn't exist, do we need to implement it?

Reply:
For RSA I don't think ring can decrypt. So we probably need to keep it as well. Rings API is totally different then what you already have so you have to change it when switching to ring (where applicable)

Reply:
As for the other components, they are completely different from those built into the ring, so it would be difficult to replace them as is.

Even if it were possible, I would like to consider it in a separate PR.

Reply:
If you don't use the p256/p384 crates I don't understand how the implementation below replaces it. Are these features implemented in ring?

modules/llrt_crypto/src/subtle/derive_bits.rs:
- p256::SecretKey::from_pkcs8_der()
- p256::SecretKey::from_pkcs8_der(public)
- p256::elliptic_curve::ecdh::diffie_hellman()
- p384::SecretKey::from_pkcs8_der()
- p384::SecretKey::from_pkcs8_der()
- p384::elliptic_curve::ecdh::diffie_hellman()

Reply:
Does this example help?
**Note that this example uses X25519, but ECDH using NIST P-256/P-384 is done exactly the same way, just substituting agreement::ECDH_P256/agreement::ECDH_P384 for agreement::X25519.**
https://docs.rs/ring/latest/ring/agreement/index.html#example

Reply:
It is likely that the implementation will be like this, but it is difficult because it is not possible to create a `privateKey` from `base_key`.

```rust
fn derive_bits(
    ctx: &Ctx<'_>,
    algorithm: &DeriveAlgorithm,
    base_key: &[u8],
    length: u32,
) -> Result<Vec<u8>> {
    match algorithm {
        DeriveAlgorithm::Edch { curve, public } => {
            let curve = match curve {
                EllipticCurve::P256 => &ring::agreement::ECDH_P256,
                EllipticCurve::P384 => &ring::agreement::ECDH_P384,
            };
            // Originally, privateKey should be created using base_key seed.
            let private_key =
                EphemeralPrivateKey::generate(curve, &SYSTEM_RANDOM.to_owned()).or_throw(ctx)?;
            let peer_public_key = UnparsedPublicKey::new(curve, &public[..]);

            let mut shared_secret = vec![];
            agree_ephemeral(private_key, &peer_public_key, |secret| {
                // TODO:
                // shared_secret.extend_from_slice(secret);
                // let _ = Ok::<(), Error>(());
            })
            .or_throw(ctx)?;

            Ok(shared_secret)
        },
        ...
```

Reply:
Let's revisit this in another PR :)